addons/l10n_id/models/account_move.py

```diff
@@ -0,0 +1,118 @@
+        ])
+        return invoices._update_payment_status()
+
+    def _update_payment_status(self):
+        paid_invoices = self._fetch_payment_status()
+        if paid_invoices:
+            return paid_invoices._register_payment()
+
+    def _fetch_payment_status(self):
+        """
+        For each invoices, we need to check the list of generated QR codes. If the QR code is not paid, but is aged of
+        more than 30m, we will discard it.
+        This will return the list of invoices for which we found a payment QR code.
+        """
+        paid_invoices = self.env['account.move']
+        for invoice in self:
+            qris_data_to_recheck = []
+            # Looping to make requests is far from ideal, but we have no choices as they don't allow getting multiple QR result at once.
+            # Ensure to loop in reverse and check from the most recent QR code.
+            for qr_invoice in reversed(invoice.qris_invoice_details):
+                status_response = invoice.partner_bank_id._qris_fetch_status(qr_invoice)
+                qris_response_data = status_response['data']
+                payment_status = qris_response_data.get('qris_status', 'unpaid')
+
+                if payment_status == "paid":
+                    paid_invoices |= invoice
+
+                    # Log a message to notify that the payment comes from QR.
+                    if 'qris_payment_customername' in qris_response_data and 'qris_payment_methodby' in qris_response_data:
+                        invoice.message_post(
+                            body=_(
+                                "This invoice was paid by %s using QRIS with the payment method %s.",
+                                qris_response_data.get('qris_payment_customername'),
+                                qris_response_data.get('qris_payment_methodby'),
+                            )
+                        )
+                    else:
+                        invoice.message_post(
+                            body=_("This invoice was paid using QRIS.")
+                        )
+                    break  # When paid we don't need to check anymore.
+                else:
+                    # The QR date is ib Jakarta time, so we ensure that we use correct timezones.
+                    qris_datetime = fields.Datetime.to_datetime(
+                        qr_invoice['qris_creation_datetime']
+                    ).replace(tzinfo=pytz.timezone('Asia/Jakarta'))
+                    jakarta_now = fields.Datetime.context_timestamp(self.with_context(tz='Asia/Jakarta'), fields.Datetime.now())
+                    # We will only reverify QR codes that have less than 30m of age, as after that they are no longer valid and will never be paid.
+                    if (jakarta_now - qris_datetime).total_seconds() < 1800:
+                        qris_data_to_recheck.append(qr_invoice)
+
+            invoice.qris_invoice_details = qris_data_to_recheck
```

Comment:
not sure if it's overkill but in this case I would try to remove all the invoice details who has aged > 30 mins while you're iterating over the `qris_invoice_details` too